### DISCUSSION TITLE
Related record selection

### DISCUSSION THREAD

**Discussion by vovpff** [YetiForceCompany/YetiForceCRM#14452] [How to] [answered]
After update to 6.1 I has noticed what something going wrong with selecting relation records. For example if I want to select Contact for Opportunity there appear modal with all contacts instead contacts related to Opportunity -> Account with switch button on top. 

I has checked file `app_data/moduleHierarchy.php`. Looks like everything perfect.

```
	'recordsListFilter' => [
		'Contacts' => [
			'HelpDesk' => ['fieldName' => 'parent_id', 'moduleName' => 'Accounts'],
			'Project' => ['fieldName' => 'linktoaccountscontacts', 'moduleName' => 'Accounts'],
			'SSalesProcesses' => ['fieldName' => 'related_to', 'moduleName' => 'Accounts'],
			'SQuoteEnquiries' => ['fieldName' => 'accountid', 'moduleName' => 'Accounts'],
		],
``` 

But what is wrong?

**Comment by rskrzypczak** [comment, marked as answer]
Yes, it doesn't work properly in version 6.1.
This has been fixed in the current version of dev.
In v6.2, this option will disappear and will be replaced with database entries.

**Reply by vovpff** [reply]
I has checked what nothing call `\App\ModuleHierarchy::getFieldsForListFilter($moduleName)` when I click to select button

Could you share temporary fix?

**Reply by rskrzypczak** [reply]
There is no temporary fix here, just a complete replacement.
https://github.com/YetiForceCompany/YetiForceCRM/commit/d841815bf80c46a066de454cd18ae9aa3ac47abd

**Reply by vovpff** [reply]
Yes. Too much amount for replacement.
